src/providers/keys/client.py

```diff
@@ -51,17 +54,18 @@ def _get_with_blockstamp(self, url: str, blockstamp: BlockStamp, params: dict |
         raise KeysOutdatedException(f'Keys API Service stuck, no updates for {self.backoff_factor * self.retry_count} seconds.')
 
     @lru_cache(maxsize=1)
-    @list_of_dataclasses(LidoKey.from_response)
-    def get_used_lido_keys(self, blockstamp: BlockStamp) -> list[dict]:
+    def get_used_lido_keys(self, blockstamp: BlockStamp) -> list[LidoKey]:
         """Docs: https://keys-api.lido.fi/api/static/index.html#/keys/KeysController_get"""
-        return cast(list[dict], self._get_with_blockstamp(self.USED_KEYS, blockstamp))
+        return list(map(lambda x: LidoKey.from_response(**x), self._get_with_blockstamp(self.USED_KEYS, blockstamp)))
 
     @lru_cache(maxsize=1)
-    def get_module_operators_keys(self, module_address: StakingModuleAddress, blockstamp: BlockStamp) -> dict:
+    def get_module_operators_keys(self, module_address: StakingModuleAddress, blockstamp: BlockStamp) -> PartiallyTypedModuleOperatorsKeys:
         """
         Docs: https://keys-api.lido.fi/api/static/index.html#/operators-keys/SRModulesOperatorsKeysController_getOperatorsKeys
         """
-        return cast(dict, self._get_with_blockstamp(self.MODULE_OPERATORS_KEYS.format(module_address), blockstamp))
+        data = cast(dict, self._get_with_blockstamp(self.MODULE_OPERATORS_KEYS.format(module_address), blockstamp))
+        data['keys'] = [LidoKey.from_response(**k) for k in data['keys']]
+        return cast(PartiallyTypedModuleOperatorsKeys, data)
```

Comment:
Cast doesn't convert dict to TypedDict. This cast creates some confusion and could lead to issues.
There are two ways to fix it:
- Write fully annotated TypedDict for this response and convert dict to `ModuleOperatorsKeys`
- Or just return dict[str, Any]